src/ring_theory/ideal/basic.lean

```diff
@@ -474,6 +474,20 @@ end pi
 
 end ideal
 
+--Move this somewhere better
+lemma bot_prime [integral_domain α] : (⊥ : ideal α).is_prime :=
+begin
+  split,
+  { intro,
+    have h1 := (ideal.eq_top_iff_one) (⊥ : ideal α),
+    rw h1 at a, have : 1 = (0 : α), tauto, simpa },
+  { intros,
+    have h1 : x * y = 0, tauto,
+    have x_or_y0 : x = 0 ∨ y = 0,
+    exact zero_eq_mul.mp (eq.symm h1),
+    tauto },
+end
```

Comment:
I also golfed the proof here. Steps:

 * `have h1 : a ↔ b := e, rw h1` is just `rw e`
 * I know that `x ∈ ⊥ ↔ x = 0`, this is called `submodule.mem_bot`, so rewrite with that everywhere
 * Now the first goal is `¬ 1 = 0`, which is called `one_ne_zero`. So `intros, apply one_ne_zero, rw [...]` works to solve the first goal.
 * Now the second goal is `x * y = 0 ⊢ x = 0 ∨ y = 0`, which is called `mul_eq_zero`, so `intros, apply mul_eq_zero, simp only [...]` works.
 * Convert the short tactical proofs to term-mode proofs.

```suggestion
⟨λ h, one_ne_zero (by rwa [ideal.eq_top_iff_one, submodule.mem_bot] at h),
 λ x y h, mul_eq_zero.mp (by simpa only [submodule.mem_bot] using h)⟩
```